app/Http/Controllers/Api/BookController.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use DB;
+use App\Model\Book;
+use App\Model\Post;
+use App\Model\Borrowing;
+use App\Model\QrCode;
+use App\Model\Rating;
+use App\Model\Comment;
+use App\Model\Favorite;
+use Carbon\Carbon;
+
+class BookController extends Controller
+{
+    /**
+     * Soft delete "book" and its relationship ("borrowing", "post", "qrcode", "comment"),
+     * Hard delete "rating" with id of book.
+     *
+     * @param Request $request request
+     * @param int     $id      id of book
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy(Request $request, $id)
+    {
+        $timeDelete = Carbon::now();
+        DB::beginTransaction();
+        try {
+            //Delete book and its favorite.
+            $book = Book::where('id', $id)->update(['deleted_at' => $timeDelete, 'updated_at' => $timeDelete]);
+            Favorite::where('favoritable_id', $id)->where('favoritable_type', Favorite::TYPE_BOOK)->update(['deleted_at' => $timeDelete, 'updated_at' => $timeDelete]);
+
+            //Delete rating, qrcode, borrowing, post.
+            Rating::where('book_id', $id)->update(['deleted_at' => $timeDelete, 'updated_at' => $timeDelete]);
+            QrCode::where('book_id', $id)->update(['deleted_at' => $timeDelete, 'updated_at' => $timeDelete]);
+            Borrowing::where('book_id', $id)->update(['deleted_at' => $timeDelete, 'updated_at' => $timeDelete]);
+
+            //Get list post ids.
+            $postsID = Post::where('book_id', $id)->pluck('id')->toArray();
+
+            //Delete posts and its favorites.
+            Post::where('book_id', $id)->update(['deleted_at' => $timeDelete, 'updated_at' => $timeDelete]);
+            Favorite::whereIn('favoritable_id', $postsID)->where('favoritable_type', Favorite::TYPE_POST)->update(['deleted_at' => $timeDelete, 'updated_at' => $timeDelete]);
+
+            //Get all comments ID of posts was deleted.
+            $commentsID = Comment::whereIn('post_id', $postsID)->pluck('id')->toArray();
+
+            //Delete all comments and its favorites.
+            Comment::whereIn('post_id', $postsID)->update(['deleted_at' => $timeDelete, 'updated_at' => $timeDelete]);
+            Favorite::whereIn('favoritable_id', $commentsID)->where('favoritable_type', Favorite::TYPE_COMMENT)->update(['deleted_at' => $timeDelete, 'updated_at' => $timeDelete]);
+            DB::commit();
+        } catch (\PDOException $e) {
```

Comment:
- you should add catch Exception, because sometime will has exception which is not a PDOException.